frontend/views/pages/DesignSystem.vue

```diff
@@ -1350,7 +1350,16 @@ export default {
       return new Promise((resolve, reject) => {
         setTimeout(resolve, 2500)
       })
+    },
+    toggleTheme () {
+      this.setTheme(this.isDarkTheme ? 'light' : 'dark')
```

Comment:
It's possible that the issue with `checkThemeState` above is due to a typo. Can you convert all these string literals to be string constants? E.g. define in some file `const THEME_LIGHT = 'light'` (same for dark) and import and use it here and elsewhere.